### ISSUE TITLE
False positive 'no-undef-class' when importing scss files using an alias path (aka using 'module-resolver')

### ISSUE THREAD

**Issue by rbf** [atfzl/eslint-plugin-css-modules#27]
Thanks for this nice plugin to `eslint`! It helps keeping the `.scss` files lean and clean!  🎉 

I'm facing an unexpected behaviour with a false positive `css-modules/no-undef-class` error. I think to have tracked it down to the case when the path of the imported `.scss` file uses an alias.

When importing the style file with the full relative path like this:

```
import cssGlobal from '../../../../assets/styles/global.scss';
```

I get the expected (long) list of classes that are defined but not used.

However, when I use the aliased version of the path (thanks to a combination of `babel-plugin-module-resolver`, `eslint-plugin-import` and `eslint-import-resolver-babel-module`) like this

```
import cssGlobal from 'styles/global.scss';
```

`css-modules` doesn't seem to find the class definitions and I get a bunch of `css-modules/no-undef-class` linting errors.

Is this a feature that is expected to work, or is something not standard in my project structure?

Any hints are greatly appreciated! Thanks! 🙂 

### `package.json`

```json
{
  "name": "my-project",
  "private": true,
  "dependencies": {
    "@rails/webpacker": "^3.0.1",
    "babel-preset-react": "^6.24.1",
    "prop-types": "^15.5.10",
    "react": "^15.6.1",
    "react-dom": "^15.6.1",
    "react-on-rails": "^9.0.2",
    "webpack-merge": "^4.1.0"
  },
  "devDependencies": {
    "babel-eslint": "^8.0.0",
    "babel-plugin-module-resolver": "^2.7.1",
    "eslint": "^3.19.0 || ^4.3.0",
    "eslint-config-airbnb": "^15.1.0",
    "eslint-import-resolver-babel-module": "^3.0.0",
    "eslint-plugin-css-modules": "^2.7.5",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-jsx-a11y": "^5.1.1",
    "eslint-plugin-react": "^7.1.0",
    "sass-lint": "^1.11.1",
    "webpack-dev-server": "^2.8.1"
  }
}

```

### `.babelrc`

```json
{
  "presets": [
    [
      "env",
      {
        "modules": false,
        "targets": {
          "browsers": "> 1%",
          "uglify": true
        },
        "useBuiltIns": true
      }
    ],
    "react"
  ],
  "plugins": [
    "syntax-dynamic-import",
    "transform-object-rest-spread",
    ["module-resolver", {
      "root": ["./client"],
      "alias": {
        "styles": "./client/app/assets/styles",
        "img": "./client/app/assets/img"
      },
      "extensions": [
        ".js",
        ".jsx",
        ".scss",
        ".css"
      ]
    }],
    [
      "transform-class-properties",
      {
        "spec": true
      }
    ]
  ]
}
```

### `.eslintrc.yml`

```yml
plugins:
  - css-modules

extends:
  - airbnb
  - plugin:css-modules/recommended

# SOURCE: https://github.com/eslint/eslint/issues/4636#issuecomment-162910216
parser: babel-eslint

# SOURCE: https://github.com/tleunen/eslint-import-resolver-babel-module/issues/25#issuecomment-242980792
settings:
  import/resolver:
    babel-module:
      extensions:
        - ".js"
        - ".jsx"
        - ".scss"
        - ".css"
```

**Comment by o2dazone** [comment]
I'm thinking you'd need to use `resolvePath` from that package

```javascript
import { resolvePath } from 'babel-plugin-module-resolver';

...

const resolvedPath = resolvePath(styleFilePath, './src', {
  "root": [ "./src"]
});

...

return { importName, styleFilePath: resolvedPath, importNode };
```

I kinda hacked it up on a fork, but the "root" from `.babelrc` is hardcoded, and haven't had much time to do a proper PR. If someone else would like to take a stab at it
